config/crd/bases/provider.giantswarm.io_kvmmachinetemplates.yaml

```diff
@@ -0,0 +1,243 @@
+                            description: 'Namespace of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/'
+                            type: string
+                          resourceVersion:
+                            description: 'Specific resourceVersion to which this reference
+                              is made, if any. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#concurrency-control-and-consistency'
+                            type: string
+                          uid:
+                            description: 'UID of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#uids'
+                            type: string
+                        type: object
+                      hostVolumes:
+                        description: HostVolumes defines the host volumes (PV) to
+                          be mounted into the workload cluster machine.
+                        items:
+                          properties:
+                            hostPath:
+                              description: HostPath is the path of the directory in
+                                the management cluster machine to be mounted into
+                                the workload cluster machine.
+                              minLength: 1
+                              type: string
+                            mountTag:
+                              description: MountTag is the value of the mount tag
+                                label which will be used to select the PV to mount
+                                to the machine.
+                              maxLength: 31
+                              minLength: 1
+                              type: string
+                          required:
+                          - hostPath
+                          - mountTag
+                          type: object
+                        nullable: true
+                        type: array
+                      providerID:
+                        description: ProviderID is the unique identifier of the machine
+                          assigned by the provider.
+                        type: string
+                      resourceRequirements:
+                        description: ResourceRequirements defines the compute and
+                          storage requirements of the machine.
+                        properties:
+                          cpu:
+                            anyOf:
+                            - type: integer
+                            - type: string
+                            description: CPU is the amount of CPU (whole or partial)
+                              reserved for the virtual machine.
+                            pattern: ^(\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))(([KMGTPE]i)|[numkMGTPE]|([eE](\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))))?$
```

Comment:
I see you use same regex than from mem and storage, but CPU has different units according to https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#resource-units-in-kubernetes, Am I right?

Reply:
Good point. These values map directly to Kubernetes container requests/limits so I took the definitions from the PodSpec here https://github.com/kubernetes/api/blob/c112aae146780fbf16b7c31dc087440b6e30dcc6/core/v1/types.go#L4962-L4989 It doesn't make much sense to have less than 1 memory, whereas 0.001 is a valid value for CPU. We actually had an issue recently with a pod with a memory limit of 500m instead of 500Mi and a better regex would have caught it. This regex for `resource.Quantity` comes from upstream Kubernetes so I think we might just have to live with it.